### DISCUSSION TITLE
New article about using object_store to add cloud-file reading to a genomics package.

### DISCUSSION THREAD

**Discussion by CarlKCarlK** [apache/arrow-rs#5381] [Show and tell]
Thanks for the help and responsiveness of the object_store team! With your help, I was able to use object_store to add cloud-file reading to bed-reader, a genomics crate (Rust) and package (Python).

I wrote up my experience in an article, [Nine Rules for Accessing Cloud Files from Your Rust Code](https://towardsdatascience.com/nine-rules-for-accessing-cloud-files-from-your-rust-code-d456c1e2ceb4).

You may also be interested in looking at companion wrapper crate called [`cloud-file`](https://crates.io/crates/cloud-file). If any of its examples, documentation, code would be useful to the object_store project, feel free to adapt/use them. (It is permissively licensed.)
I think the three examples--if written completely with the object_store API--could be especially useful to object_store users. The examples are:
* Sequentially read the bytes of a HTTP file, counting new line characters
* Sequentially read the text lines of an HTTP file, returning the nth line.
* Read selected regions of an HTTP file to tabulate and report on the most frequent character bigrams (Also, limit bytes read per request and the number of concurrent requests.)

Thanks again for creating the powerful object_store crate! -- Carl

**Comment by tustvold** [comment]
Thank you, I'll be sure to take a look in the next few days. FWIW documentation PRs are always welcome and appreciated
